### PR TITLE
Compound field count

### PR DESCRIPTION
**Pull request by thebailers**

### CODE REVIEW

File: app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/date.scala.html [hunk 1]
```diff
@@ -30,24 +38,66 @@
...
         <div class="govuk-date-input__item">
             <div class="govuk-form-group">
                 <label class="govuk-label govuk-date-input__label" for="@key-@field">
                 @messages(List((key :+ field).mkString("."),field))
                 </label>
                 <input
-                class="govuk-input govuk-date-input__input govuk-input--width-@if(field=="year"){4}else{2} @if(errors.definedAt(field)) {govuk-input--error}"
+                class="form-control govuk-input govuk-date-input__input govuk-input--width-@if(field=="year"){4}else{2}
+                    @errors.values.headOption.map { er =>
+                        @if(er.head.msg.contains(field || "not-a-date")) {govuk-input--error}
```
Comment (adamdye-hmrc) [inline app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/date.scala.html]: This needs to be `@if(er.head.msg.contains(field) || (er.head.msg === "not-a-date"))`
I pushed that change but it seems to have got lost.

Comment (adamdye-hmrc) [reply app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/date.scala.html]: There's an example on line 91

File: app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/string.scala.html [hunk 2]
```diff
@@ -35,20 +35,30 @@
...
-        </span>
-    }
+<div class="form-group@if(errors.definedAtRoot){ form-group-error}">
+    @standard_field(key, errors, "string", messages) {
+    
+        @if(currencyInput){
+            <span class="input-currency">
+        }
+
+        @defining((messages.get(s"${key.mkString(".")}.hint").isDefined, errors.definedAtRoot) match {
```
Comment (adamdye-hmrc) [inline app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/string.scala.html]: Couldn't this be in a helper?

File: build.sbt [hunk 3]
```diff
@@ -49,3 +49,7 @@ libraryDependencies ++= Seq(
 )
 
 scalacOptions += "-P:silencer:pathFilters=.*views.*;.*routes"
+
+// If you're doing a PR, or someone has asked you to merge this into master
+// politely decline - snapshots are for development only. 
+resolvers += Resolver.sonatypeRepo("snapshots")
```
Comment (adamdye-hmrc) [inline build.sbt]: I politely decline :)

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -3,7 +3,9 @@ import sbt._
 
 object AppDependencies {
 
-  val uniformVersion = "4.5.1"
+  // If you're doing a PR, or someone has asked you to merge this into master
```
Comment (adamdye-hmrc) [inline project/AppDependencies.scala]: Why is this comment here?

File: app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/fieldsurround.scala.html [hunk 5]
```diff
@@ -14,63 +14,86 @@
...
+      </p>
+    }
+    
+    @*
+    * Will need to query if the content is a compound or standard field 
+    * so that we know how to wrap?
+    *@
+    @content
+
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (thebailers) [reply app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/fieldsurround.scala.html]: Removed

File: app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/date.scala.html (hunk 1 above)
Comment (thebailers) [reply app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/date.scala.html]: Updated

File: app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/string.scala.html (hunk 2 above)
Comment (adamdye-hmrc) [reply app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/string.scala.html]: Will fix in refactor

Comment (thebailers) [reply app/uk/gov/hmrc/disguisedremunerationfrontend/views/uniform/string.scala.html]: Added todo comment to refactor this

File: project/AppDependencies.scala (hunk 4 above)
Comment (adamdye-hmrc) [reply project/AppDependencies.scala]: I don't think it's meant to be here, can you remove it?

Comment (thebailers) [reply project/AppDependencies.scala]: Removed

**Pull request merged**